client/client_earthly.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// WithAdditionalHeaders adds metadata to all outgoing gRPC calls. It acts just like metadata.Pairs does, and calls *will* fail if
+// you provide an odd number of items. Values should be in alternating key-value order - ("k1", "v1", "k1", "v2", "k2", "v3").
+func WithAdditionalHeaders(kv ...string) ClientOpt {
```

Comment:
It might be helpful to avoid using the word `header`, since a user might confuse this function with a function that changes http(s) headers, perhaps it could be draw inspiration from grpc's `AppendToOutgoingContext` function?

maybe:

```suggestion
func WithAdditionalMetadataContext(kv ...string) ClientOpt {
```